code/modules/mob/living/carbon/human/species.dm

```diff
@@ -612,14 +617,28 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 							hair_overlay.color = "#" + hair_color
 					else
 						hair_overlay.color = "#" + H.hair_color
+
+					//Gradients
+					grad_style = H.grad_style
+					grad_color = H.grad_color
+					if(grad_style && grad_style != "None")
+						var/datum/sprite_accessory/gradient = GLOB.hair_gradients_list[grad_style]
+						var/icon/temp = new/icon("icon" = gradient.icon, "icon_state" = gradient.icon_state)
+						var/icon/temp_hair = new/icon("icon" = hair_file, "icon_state" = hair_state)
+						temp.Blend(temp_hair, ICON_ADD)
```

Comment:
cache the result of this for the gradient style, colors, and hair icon used, so as to avoid repeated Blend() calls, which are expensive